test/integration/ingress.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"istio.io/manager/test/util"
+)
+
+type ingress struct {
+	*infra
+	accessMu   sync.Mutex
+	accessLogs map[string][]string
+}
+
+const (
+	ingressServiceName = "istio-ingress-controller"
+)
+
+func (t *ingress) String() string {
+	return "ingress controller"
+}
+
+func (t *ingress) setup() error {
+	if !t.Ingress {
+		return nil
+	}
+	t.accessLogs = make(map[string][]string)
+	for app := range t.apps {
+		t.accessLogs[app] = make([]string, 0)
+	}
+
+	// setup ingress resources
+	if err := util.Run(fmt.Sprintf("kubectl -n %s create secret generic ingress "+
```

Comment:
We could commit the secret resource with-line key/cert data to reduce the number of test file dependencies, e.g.

```bash
kubectl create secret generic ingress \
      --from-file=tls.key=test/integration/cert.key \
      --from-file=tls.crt=test/integration/cert.crt --dry-run -o yaml > test/integration/testdata/ingress-secret.yaml
```

Then we'd just `kubectl apply` like we do for ingress.yaml